packages/vertexai/src/types/error.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GenerateContentResponse } from './responses';
+
+/**
+ * Details object that may be included in an error response.
+ *
+ * @public
+ */
+export interface ErrorDetails {
+  '@type'?: string;
+
+  /** The reason for the error. */
+  reason?: string;
+
+  /** The domain where the error occured. */
+  domain?: string;
+
+  /** Additonal metadata about the error. */
+  metadata?: Record<string, unknown>;
+
+  /** Any other relevant information about the error. */
+  [key: string]: unknown;
+}
+
+/**
+ * Details object that contains data originating from a bad HTTP response.
+ *
+ * @public
+ */
+export interface CustomErrorData {
+  /** HTTP status code of the error response. */
+  status?: number;
+
+  /** HTTP status text of the error response. */
+  statusText?: string;
+  
+  /** Response from a {@link GenerateContentRequest} */
+  generateContentResponse?: GenerateContentResponse;
```

Comment:
I'd like to keep this as "response" so we don't have to create a new field if we want to attach a response for another method in the future. Also currently from the user's perspective, they could get this on a chat.sendMessage error (because chat.sendMessage calls generateContent under the hood) and they might be confused why their error has a generateContentResponse field when they never called generateContent as far as they know.

If it encompasses more types of responses in the future we can do a union (`|`) here with whatever new interface we add.

Reply:
Done